database/src/pool/postgres.rs

```diff
@@ -1970,6 +1912,223 @@ where
             .context("Failed to mark benchmark job as completed")?;
         Ok(())
     }
+
+    async fn get_status_page_data(&self) -> anyhow::Result<PartialStatusPageData> {
+        let max_completed_requests = 7;
+        let in_progress_requests_query = format!(
+            "
+            SELECT {BENCHMARK_REQUEST_COLUMNS}
+            FROM benchmark_requests
+            WHERE status = '{BENCHMARK_REQUEST_STATUS_IN_PROGRESS_STR}'
+        "
+        );
+        // Gets requests along with how long the request took (latest job finish
+        // - earliest job start) and associated errors with the request
+        let completed_requests_query = format!(
+            "
+            WITH completed AS (
+                SELECT
+                    {BENCHMARK_REQUEST_COLUMNS}
+                FROM
+                    benchmark_request
+                WHERE
+                    status = '{BENCHMARK_REQUEST_STATUS_COMPLETED_STR}'
+                ORDER BY
+                    completed_at
+                DESC LIMIT {max_completed_requests}
+            ), jobs AS (
+                SELECT
+                    completed.tag,
+                    job_queue.started_at,
+                    job_queue.completed_at
+                FROM
+                    job_queue
+                LEFT JOIN completed ON job_queue.request_tag = completed.tag
+            ), stats AS (
+                SELECT
+                    tag,
+                    MAX(jobs.completed_at - jobs.started_at) AS duration
+                FROM
+                    jobs
+                GROUP BY
+                    tag
+            ), artifacts AS (
+                SELECT
+                    artifact.id,
+                    \"name\"
+                FROM
+                    artifact
+                LEFT JOIN completed ON artifact.name = completed.tag
+            ), errors AS (
+                SELECT
+                    artifacts.name AS tag,
+                    ARRAY_AGG(error) AS errors
```

Comment:
Huh, interesting. The error count should be low, so normally I would just do a join and call it a day, but if this works, then why not.

Reply:
Something to note on the errors is that it needs a unique "krate", aid pairing. Which if we are using for arbitrary error collection may mean we need think about this